### ISSUE TITLE
Connect with OneNote - RESULT "forbidden request"

### ISSUE THREAD

**Issue by nhwCoder** [anilkumarum/one-clipper#4]
When I try to connect to a "MS365 Business S" account (company account), I get after sucessfull login with username+password

## Error message
forbidden request

## Mesage in URL
- The+application+'oneClipper'+asked+for+scope+'Notes.Create,Notes.ReadWrite,Notes.ReadWrite.All,wl.offline_access'+that+doesn't+exist+on+the+resource
- Contact+the+app+vendor.

------
## URL
https://cloud.oneclipper.noterail.co/authorize/connect-one-note?error=invalid_client&error_description=AADSTS650053%3a+The+application+%27oneClipper%27+asked+for+scope+%27Notes.Create%2cNotes.ReadWrite%2cNotes.ReadWrite.All%2cwl.offline_access%27+that+doesn%27t+exist+on+the+resource+%2700000003-0000-0000-c000-000000000000%27.+Contact+the+app+vendor.+Trace+ID%3a+d88883f6-2ee4-4f0d-94ef-923ae0da1a00+Correlation+ID%3a+af486929-64ce-4f9c-b306-175cce4d49d8+Timestamp%3a+2024-10-09+12%3a52%3a27Z#

## URL - Decoded
https://cloud.oneclipper.noterail.co/authorize/connect-one-note?error=invalid_client&error_description=AADSTS650053:+The+application+'oneClipper'+asked+for+scope+'Notes.Create,Notes.ReadWrite,Notes.ReadWrite.All,wl.offline_access'+that+doesn't+exist+on+the+resource+'00000003-0000-0000-c000-000000000000'.+Contact+the+app+vendor.+Trace+ID:+d88883f6-2ee4-4f0d-94ef-923ae0da1a00+Correlation+ID:+af486929-64ce-4f9c-b306-175cce4d49d8+Timestamp:+2024-10-09+12:52:27Z#
´
## Versions
OneClipper Version: 0.0.29
MS Edge: Version 129.0.2792.79 (Official Build) (64-Bit)

**Comment by anilkumarum** [comment]
Namaste

I think - `Notes.ReadWrite.All` API permission creates this problem. Your company doesn't give access to this scope `Notes.ReadWrite.All`.

Microsoft explains:
`Notes.ReadWrite.All` : Allow the application to view and modify the contents of all OneNote notebooks and sections that the signed-in user has access to. It cannot access password protected sections.

I will test extension without this permission and then post an update.

**Comment by kendoori** [comment]
I have the same problem... I'm a corporate Windows 365 user and don't have the ability to request changes at a corporate policy level to permissions.

**Comment by pullinblanks** [comment]
Same problem here. Hopefully there's a way to fix this!

**Comment by anilkumarum** [comment]
I removed `Notes.ReadWrite.All` from API oAuth authorization request.
Please test if it fixes this problem or not.
I don't have a business account so  I cannot test this issue from my side.
I am waiting for your reply.

**Comment by nhwCoder** [comment]
#### QUESTION
* WHAT ist the technical reason you make a call from your website URL https://cloud.oneclipper.noterail.co/authorize/connect-one-note to authorize? (and not only from my webbrowser to OneNote cloud)

#### Test with  "MS365 Business S" account
- Click button "Connect With Onenote"
- MS PopUp with "Try to connect" opens
- Error message in webbrowser: "Forbidden request" (= see F12 console messages below)

#### Info
* MSEdge newest version 130.0.2849.52
* oneClipper Version 0.0.28

#### F12 Console messages
Uncaught ReferenceError: MutationEvent is not defined 
    at t (content.js:2:412)
    at content.js:8:33
    at content.js:77:246
    at content.js:77:251
*===> INFO:  Just from Tampermonkey, that this page is ignored by Tampermonkey*

Failed to load resource: the server responded with a status of 403 ()        connect-one-note:1 
*===> INFO:  From the following URL request*
https://cloud.oneclipper.noterail.co/authorize/connect-one-note?error=invalid_client&error_description=AADSTS650053%3a+The+application+%27oneClipper%27+asked+for+scope+%27Notes.Create%2cNotes.ReadWrite%2cwl.offline_access%27+that+doesn%27t+exist+on+the+resource+%2700000003-0000-0000-c000-000000000000%27.+Contact+the+app+vendor.+Trace+ID%3a+70a1628a-0491-4149-ba70-37a5ad491a00+Correlation+ID%3a+96b6bfec-c45c-46e1-b677-e3bafe3c80d4+Timestamp%3a+2024-10-24+09%3a35%3a28Z

**Comment by anilkumarum** [comment]
Reason for making a call from your website URL
We cannot  expose API secret key (provided by Microsoft) to browsers that why I need to make requests from server-side.

**Comment by anilkumarum** [comment]
oneClipper asked for 3 scope:
1. Notes create -> need to create new note (cannot remove)
2. Notes ReadWrite -> manadatory scope (cannot remove)
3. wl.offline_access -> without this scope, cannot refresh OAuth token that expires in 1hr (cannot remove)

Can you confirm with your IT team?
Do you have access to use One Note  API

I wait for other user responses.
Then I will create a business account and check which scope will be blocked by business account admin.

**Comment by kendoori** [comment]
Is the latest in the Chrome Web store? I have a business account and just
tired to enable it, and ran into the "forbidden issue still.

**Comment by anilkumarum** [comment]
@kendoori This issue will be addressed on the server side. I’ll work on reproducing the bug to resolve it. In the meantime, version v0.0.30 has been submitted for verification and should be available within 1-2 days.

**Comment by nhwCoder** [comment]
>I don't have a business account so I cannot test this issue from my side.
Could you reproduce the bug?

**Comment by nhwCoder** [comment]
#### Tested with Version 0.0.34.
* The same bug still exists with a MS365 business account.

**Comment by anilkumarum** [comment]
I posted this question on the Microsoft Community Forum on November 11, but I haven't received any replies yet. 
[https://learn.microsoft.com/en-us/answers/questions/2117865/(onenote-error)-required-scope-doesnt-exist-for-bu](https://learn.microsoft.com/en-us/answers/questions/2117865/(onenote-error)-required-scope-doesnt-exist-for-bu)

I also asked the same question on StackOverflow, hoping for a response. I'll provide updates here once I get an answer.
[https://stackoverflow.com/questions/79288092/required-scope-doesnt-exist-error-caused-for-onenote-business-account-user](https://stackoverflow.com/questions/79288092/required-scope-doesnt-exist-error-caused-for-onenote-business-account-user)

**Comment by anilkumarum** [comment]
The 'Forbidden Request' issue when linking to OneNote (Office 365) has been resolved. With the latest update (v0.0.18) and chrome extension (0.0.35), we’ve added support for Office 365 accounts. Extension should auto-update. 
Please select your one note account type:  Personal or Work/Office 365 before hitting Connect button.
Please give it another try, and let us know if you encounter any further issues.

**Comment by anilkumarum** [comment]
I think Issue is fixed now for almost all users. If not, please create new issue
